spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -348,25 +344,25 @@ private void postProcessAbstractEndpoint(Object bean) {
 		if (bean instanceof IntegrationConsumer) {
 			IntegrationConsumer integrationConsumer = (IntegrationConsumer) bean;
 			MessageHandler handler = integrationConsumer.getHandler();
-			if (handler instanceof org.springframework.integration.support.management.MessageHandlerMetrics) {
-				org.springframework.integration.support.management.MessageHandlerMetrics messageHandlerMetrics =
-						(org.springframework.integration.support.management.MessageHandlerMetrics) extractTarget(handler);
-				registerHandler(messageHandlerMetrics);
-				this.handlers.add(messageHandlerMetrics);
-				this.runtimeBeans.add(messageHandlerMetrics);
+			if (handler instanceof AbstractMessageHandler) {
```

Comment:
I think we don't need this `if`. The logic below covers everything we needed.
And having a `MessageHandler` as a proxy won't give us a positive match with this `if` anyway.

Reply:
Yeah - I went through a lot of iterations with this; I thought I had cleaned all these up. Good catch.